docs/sdk/serverless-signing/zero-to-hero.md

```diff
@@ -0,0 +1,181 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Zero to Hero
+
+## Prerequisites
+
+- Familiarity with JavaScript
+
+## Overview
+In this section we're gonna cover an end-to-end example from minting a PKP through the ContractsSDK, setting its scope & signing a message with it inside a Lit Action.
+
+Installed the latest contracts-sdk on `cayenne`
+
+```bash
+yarn install @lit-protocol/contracts-sdk@cayenne
+```
+
+## Set up the controller
+
+To initialize a LitContracts client you need an Eth Signer. This can be a standard Ethers wallet or also a `PKPEthersWallet` (more info on the later [here](../authentication/session-sigs/auth-methods/add-remove-auth-methods)). But here, we're gonna use the standard Ethers wallet.
+
+## Initialize the ContractsSDK
+
+We're using the ContractsSDK for the minting the PKP & interacting with it (updating the scopes). So let's initialize it
+
+```jsx
+const contractClient = new LitContracts({
+  signer: wallet,
+});
+
+await contractClient.connect();
+```
+
+**Note:** The default LitNetwork for the ContractsSDK is Cayenne so there's no need to set the network explicitly.
+
+## Mint the PKP
+
+Now that we've the ContractsSDK initialized we're ready to mint the PKP using it. Since we want to allow out PKP to sign messages we have to add Auth Method scopes for `SigningAnything` & `OnlySignMessages` as below otherwise you'll get an error stating that the PKP isn't authorized to sign.
+
+**Note:** You're gonna need an AuthSig for setting the `authMethod`. In the browser you can use `checkAndSignAuthMessage` or use [Hot wallet signing](https://developer.litprotocol.com/v3/support/faq/#1-cant-use-checkandsignauthmessage-in-a-backend-project) in the backend.
+
+### Get the Latest Eth Blockhash
+
+Since the ContractsSDK doesn't proveid you with the latest Eth Blockhash which is supposed to be the nonce for our AuthSig signed message we have to use the LitNodeClient to get that.
```

Comment:
can we link to the docs section where the nonce requirement is defined? https://developer.litprotocol.com/v3/sdk/authentication/auth-sig/#obtaining-an-authsig-on-the-server-side